src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/AbstractBuildEventValueProvider.AbstractBuildEventHelper.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Construction;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties.InterceptedProjectProperties
+{
+    internal abstract partial class AbstractBuildEventValueProvider
+    {
+        public abstract class AbstractBuildEventHelper
+        {
+            private const string _execTask = "Exec";
+            private const string _command = "Command";
+
+            protected AbstractBuildEventHelper(string buildEvent,
+                   string targetName,
+                   Func<ProjectTargetElement, string> getTarget,
+                   Action<ProjectTargetElement> setTargetDependencies)
+            {
+                BuildEvent = buildEvent;
+                TargetName = targetName;
+                GetTarget = getTarget;
+                SetTargetDependencies = setTargetDependencies;
+            }
+
+            private Func<ProjectTargetElement, string> GetTarget { get; }
+            private Action<ProjectTargetElement> SetTargetDependencies { get; }
+            private string BuildEvent { get; }
+            private string TargetName { get; }
+
+            public string GetProperty(ProjectRootElement projectXml)
+            {
+                // Check if project file already has props in place for this.
+                var result = TryGetFromProperty(projectXml);
+                if (result.success)
+                {
+                    return result.property;
+                }
+
+                // Check if build events can be found in targets
+                return GetFromTargets(projectXml);
+            }
+
+            public void SetProperty(string unevaluatedPropertyValue, ProjectRootElement projectXml)
```

Comment:
My logic is:
- if the pre-build or post-build target has multiple exec tasks  there is no way for us to know which one to use so we select the first one.  That way we can at least make the behavior observably deterministic
- If the user deletes the text of a post or pre build event (or sets it to just spaces and tabs) we remove the target only if there is only one exec task.  This ensures that we do not remove targets that have custom logic written by the developer.

Both cases, the user has written something custom so we err on the side of not changing things.  What do we need to communicate to the user in the scenario:
> But we need to handle the other scenario of having more than one task because this behavior silently passes without telling the user something was wrong.